src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The `useSelector.withTypes<RootState>()` is incorrect. You should use `useSelector` with TypeScript generics like this: `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;`.